client.js

```diff
@@ -121,27 +121,36 @@ RingpopClient.prototype._request = function _request(opts, endpoint, head, body,
         return;
     }
 
-    this.subChannel.waitForIdentified({
-        host: opts.host
-    }, function onIdentified(err) {
-        if (err) {
-            callback(err);
-            return;
-        }
+    try {
+        request();
+    } catch(e) {
+        callback(e);
+        return;
+    }
```

Comment:
Hi Raynos, tchannel crashes when the hostport is malformed. In a rare case, where I tcurled with a malformed hostport field, I have crashed the entire cluster. This specific case doesn't happen anymore after we hardened the join mechanics of ringpop but I still want to strengthen this. Do you know another way?